content/blog/2016/2016-05-10-jenkins-20-vjam.adoc

```diff
@@ -0,0 +1,355 @@
+</center>
+++++
+
+
+The event was really interesting for me, and I hope informative for those who
+participated in the live stream and Q&A session. I look forward to hosting more
+Virtual JAM events in the future, and I hope you will
+*link:http://www.meetup.com/Jenkins-online-meetup/[join us!]*
+
+=== Questions and Answers
+
+Below are a collection of questions and answers, that were posed during the
+Virtual JAM. Many of these were answered during the course of the sessions, but
+for posterity all are included below.
+
+==== Pipeline
+
+____
+What kind of DSL is used behind pipeline as code? Groovy or allow freely use
+different languages as a user prefer?
+____
+
+Pipeline uses a Groovy-based domain specific language.
+
+____
+How do you test your very own pipeline DSL?
+____
+
+Replay helps in testing/debugging while creating pipelines and at the branch
+level. There are some ideas which link:https://github.com/jglick[Jesse Glick]
+has proposed for testing `Jenkinsfile` and Pipeline libraries captured in
+link:https://issues.jenkins-ci.org/browse/JENKINS-33925[JENKINS-33925].
+
+____
+Isn't "Survive Jenkins restart" exclusive to Jenkins Enterprise?
+____
+
+No, this feature does not need CloudBees Jenkins Enterprise. All features shown
+during the virtual JAM are free and open source. CloudBees' Jenkins Enterprise
+product does support restarting from a specified stage however, and that is not
+open source.
+
+____
+How well is jenkins 2.0 integrate with github for tracking job definitions?
+____
+
+Using the
+link:https://wiki.jenkins-ci.org/display/JENKINS/GitHub+Organization+Folder+Pluginp[GitHub
+Organization Folder plugin], Jenkins can automatically detect a `Jenkinsfile` in
+source repositories to create Pipeline projects.
+
+
+____
+Please make the ability for re-run failed stages Open Source too :)
+____
+
+This has been passed on to our friends at CloudBees for consideration :)
+
+____
+If `Jenkinsfile` is in the repo, co-located with code, does this mean Jenkins can
+auto-detect new jobs for different branches?
+____
+
+This is possible using the
+link:/doc/pipeline/#creating-multibranch-pipelines[Pipeline Multibranch plugin].
+
+
+____
+What documentation sources are there for Pipeline?
+____
+
+Our link:/doc/[documentation section] contains a number of pagesaround Pipeline.
+There is also additional documentation and examples in the plugin's
+link:https://github.com/jenkinsci/pipeline-plugin[git repository] and the
+link:https://github.com/jenkinsci/pipeline-examples[jenkinsci/pipeline-examples]
+repository. (contributions welcome!)
+
+____
+Where we can find the DSL method documentation?
+____
+
+There is generated link:/doc/pipeline/steps[documentation on jenkins.io] which
+incldues steps from all public plugins. Inside of a running Jenkins instance,
+you can also navigate to
+link:http://localhost:8080/workflow-cps-snippetizer/dslReference[localhost:8080/workflow-cps-snippetizer/dslReference]
```

Comment:
JENKINS_URL or at least localhost:8080/jenkins/.... (to address the default installer behavior)